crates/jstz_node/src/services/logs/mod.rs

```diff
@@ -1,85 +1,204 @@
+                        current_line = lines.next_line() => {
+                            if let Ok(Some(line_str)) = current_line {
+                                if let Some(line) = Self::parse_line(&line_str) {
+                                        let _ = db.flush(&line).await.map_err(|e|
+                                            {
+                                                println!("Failed to flush log to database: {:?}", e.to_string());
+                                            }
+                                        );
 
-    async fn tail_file(
-        path: String,
-        broadcaster: Arc<Broadcaster>,
-        stop_signal: CancellationToken,
-    ) -> std::io::Result<()> {
-        let file = TailedFile::init(&path).await?;
-        let mut lines = file.lines();
-        loop {
-            tokio::select! {
-                line = lines.next_line() => {
-                    if let Ok(Some(msg)) = line {
-                        if msg.starts_with(LOG_PREFIX) {
-                            let log = LogRecord::try_from_string(&msg[LOG_PREFIX.len()..])
-                                .expect("Failed to parse log record from string");
-                            broadcaster.broadcast(&log.contract_address, &msg[LOG_PREFIX.len()..]).await;
+                                        // Steram the log
+                                        if let Line::Js(log) = line {
+                                            broadcaster_arc
+                                                .broadcast(&log.contract_address, &line_str[LOG_PREFIX.len()..])
+                                                .await;
+                                        }
+                                }
+                            }
+                        },
+                        _ = stop_signal.cancelled() => {
+                            // The stop signal has been triggered.
+                            break;
                         }
                     }
-                },
-                _ = stop_signal.cancelled() => {
-                    // The stop signal has been triggered.
-                    break;
                 }
-            }
+
+                Ok(())
+            },
+        );
+
+        Ok((broadcaster, pool, tail_file_handle))
+    }
+
+    fn parse_line(line: &str) -> Option<Line> {
+        if ![LOG_PREFIX, REQUEST_START_PREFIX, REQUEST_END_PREFIX]
+            .iter()
+            .any(|pre| line.starts_with(pre))
+        {
+            return None;
         }
-        Ok(())
+
+        if line.starts_with(LOG_PREFIX) {
+            return LogRecord::try_from_string(&line[LOG_PREFIX.len()..]).map(Line::Js);
+        }
+
+        let request_prefix = if line.starts_with(REQUEST_START_PREFIX) {
+            REQUEST_START_PREFIX
+        } else {
+            REQUEST_END_PREFIX
+        };
+
+        RequestEvent::try_from_string(&line[request_prefix.len()..]).map(Line::Request)
```

Comment:
I'd avoid checking the initial negation at the start and just have `if line.starts_with` -- more maintainable (only 1 place to modify prefix checking) + fewer string comparisons